java/org/contikios/cooja/Cooja.java

```diff
@@ -863,6 +863,49 @@ public Plugin getPlugin(String classname) {
     return null;
   }
 
+  /**
+   * Returns a started plugin that is assignable by the given class, if any.
+   *
+   * @param type a class specifying the type of plugin
+   * @return Plugin instance
+   */
+  public <T extends Plugin> T getPlugin(Class<T> type) {
+    for (Plugin p: startedPlugins) {
+      if (type.isInstance(p)) {
+        return type.cast(p);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns all started plugins that are assignable by the given class, if any.
+   *
+   * @param type a class specifying the type of plugin
+   * @return A list of plugin instances
+   */
+  public <T extends Plugin> List<T> getAllPlugins(Class<T> type) {
+    return getAllPlugins(type, null);
+  }
+
+  /**
+   * Returns all started plugins that are assignable by the given class, if any,
+   * except for the specified plugin.
+   *
+   * @param type a class specifying the type of plugin
+   * @param exclude exclude this plugin if not null
+   * @return A list of plugin instances
+   */
+  public <T extends Plugin> List<T> getAllPlugins(Class<T> type, T exclude) {
```

Comment:
I'm not sure why one needs to exclude something, but the exclude should probably be of type `Set<T>`.

Reply:
Shortcut for a plugin to lookup up all plugins of the same type as itself, except for itself. I solved it in another way and removed it. Thanks!